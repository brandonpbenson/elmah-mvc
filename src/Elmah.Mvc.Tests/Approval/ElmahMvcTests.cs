using System.Text.RegularExpressions;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace Elmah.Mvc.Tests.Approval
{
    [UseReporter(typeof(DiffReporter))]
    public class ElmahMvcTests
    {
        private readonly string ElmahMvcAppUrl = "http://localhost:49800";

        [Test]
        public void should_open_elmah_page(string path)
        {
            // do
            var url = ElmahMvcAppUrl + path;

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_stylesheets()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/stylesheet";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));            
        }

        [Test]
        public void should_open_rss()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/rss";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));            
        }

        [Test]
        public void should_open_digest_rss()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/digestrss";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_details()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/detail?id=d5b2db50-0d5b-4c1d-a4dc-30324081191a";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_not_found_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/detail";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_json_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/json?id=d6c13c5b-f57f-41e0-b23d-6de2dfa5bba7";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_xml_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/xml?id=d6c13c5b-f57f-41e0-b23d-6de2dfa5bba7";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        private string GetContent(string url)
        {
            return HackServertTime(ApprovalTests.Asp.AspApprovals.GetUrlContents(url));
        }

        private string HackServertTime(string content)
        {
            return new Regex("Server time is \\d+:\\d+:\\d+").Replace(content, string.Empty);
        }
    }
}
