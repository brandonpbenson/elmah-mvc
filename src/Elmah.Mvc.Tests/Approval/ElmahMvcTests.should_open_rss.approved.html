<rss xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" version="0.91">
  <channel>
    <title>Error log of / on SEEKEY-NOTE</title>
    <link>http://localhost:49800/admin/elmah</link>
    <description>Log of recent errors</description>
    <language>en</language>
    <item>
      <title>Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).</title>
      <description>An error of type System.FormatException occurred. Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).</description>
      <pubDate>Fri, 08 Jun 2012 15:39:34 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=f034b38b-a07a-4acd-b736-9decea0f707a</link>
    </item>
    <item>
      <title>Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).</title>
      <description>An error of type System.FormatException occurred. Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).</description>
      <pubDate>Fri, 08 Jun 2012 15:39:22 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=e5a1477f-30f3-41d7-b9b2-d4d9b3751e4d</link>
    </item>
    <item>
      <title>Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).</title>
      <description>An error of type System.FormatException occurred. Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).</description>
      <pubDate>Fri, 08 Jun 2012 15:39:10 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=52abe1f2-df17-42d1-b487-89600077f721</link>
    </item>
    <item>
      <title>Missing error identifier specification.</title>
      <description>An error of type Elmah.ApplicationException occurred. Missing error identifier specification.</description>
      <pubDate>Fri, 08 Jun 2012 15:38:57 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=d6c13c5b-f57f-41e0-b23d-6de2dfa5bba7</link>
    </item>
    <item>
      <title>Missing error identifier specification.</title>
      <description>An error of type Elmah.ApplicationException occurred. Missing error identifier specification.</description>
      <pubDate>Fri, 08 Jun 2012 15:38:37 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=1d13edd6-234a-488c-9d57-5fefb52f5a4b</link>
    </item>
    <item>
      <title>Missing error identifier specification.</title>
      <description>An error of type Elmah.ApplicationException occurred. Missing error identifier specification.</description>
      <pubDate>Fri, 08 Jun 2012 15:37:37 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=3293aecd-e8f6-4cee-9d46-41fb060de9b1</link>
    </item>
    <item>
      <title>Missing error identifier specification.</title>
      <description>An error of type Elmah.ApplicationException occurred. Missing error identifier specification.</description>
      <pubDate>Fri, 08 Jun 2012 15:36:44 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=a8d8e5be-0d48-4091-9445-5556dc0b7e61</link>
    </item>
    <item>
      <title>Missing error identifier specification.</title>
      <description>An error of type Elmah.ApplicationException occurred. Missing error identifier specification.</description>
      <pubDate>Fri, 08 Jun 2012 15:36:36 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=4cb32a93-cb99-4517-9627-625fb2f74dc0</link>
    </item>
    <item>
      <title>Missing error identifier specification.</title>
      <description>An error of type Elmah.ApplicationException occurred. Missing error identifier specification.</description>
      <pubDate>Fri, 08 Jun 2012 15:36:33 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=2072c375-6828-458a-8f6a-5817425c970a</link>
    </item>
    <item>
      <title>A public action method 'st' was not found on controller 'Elmah.Mvc.Web.Areas.Admin.Controllers.ElmahController'.</title>
      <description>An error of type System.Web.HttpException occurred. A public action method 'st' was not found on controller 'Elmah.Mvc.Web.Areas.Admin.Controllers.ElmahController'.</description>
      <pubDate>Fri, 08 Jun 2012 15:19:02 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=d5b2db50-0d5b-4c1d-a4dc-30324081191a</link>
    </item>
    <item>
      <title>A public action method 'st' was not found on controller 'Elmah.Mvc.Web.Areas.Admin.Controllers.ElmahController'.</title>
      <description>An error of type System.Web.HttpException occurred. A public action method 'st' was not found on controller 'Elmah.Mvc.Web.Areas.Admin.Controllers.ElmahController'.</description>
      <pubDate>Fri, 08 Jun 2012 15:19:02 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=79f1c6bf-e3e4-4c31-b0ab-7d99c95cd83a</link>
    </item>
    <item>
      <title>The controller for path '/not-found' was not found or does not implement IController.</title>
      <description>An error of type System.Web.HttpException occurred. The controller for path '/not-found' was not found or does not implement IController.</description>
      <pubDate>Fri, 08 Jun 2012 14:34:15 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=24fa4f44-aed0-4c46-920e-e946f623d3ff</link>
    </item>
    <item>
      <title>The controller for path '/not-found' was not found or does not implement IController.</title>
      <description>An error of type System.Web.HttpException occurred. The controller for path '/not-found' was not found or does not implement IController.</description>
      <pubDate>Fri, 08 Jun 2012 14:34:14 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=67b4e325-9342-4e71-8f83-9d700cb17557</link>
    </item>
    <item>
      <title>The method or operation is not implemented.</title>
      <description>An error of type System.NotImplementedException occurred. The method or operation is not implemented.</description>
      <pubDate>Fri, 08 Jun 2012 14:34:05 GMT</pubDate>
      <link>http://localhost:49800/admin/elmah/detail?id=53e3c86e-85f8-4b00-b98c-2354808b15b8</link>
    </item>
  </channel>
</rss>